tests/modules/lint.py

```diff
@@ -105,3 +106,110 @@ def test_modules_lint_patched_modules(self):
     assert len(module_lint.failed) == 1
     assert len(module_lint.passed) > 0
     assert len(module_lint.warned) >= 0
+
+
+# A skeleton object with the passed/warned/failed list attrs
+# Use this in place of a ModuleLint object to test behaviour of
+# linting methods which don't need the full setup
+class MockModuleLint:
+    def __init__(self):
+        self.passed = []
+        self.warned = []
+        self.failed = []
+
+
+PROCESS_LABEL_GOOD = (
+    """
+    label process_high
+    cpus 12
+    """,
+    1,
+    0,
+    0,
+)
+PROCESS_LABEL_NON_ALPHANUMERIC = (
+    """
+    label a:label:with:colons
+    cpus 12
+    """,
+    0,
+    1,
+    0,
+)
+PROCESS_LABEL_GOOD_CONFLICTING = (
+    """
+    label process_high
+    label process_low
+    cpus 12
+    """,
+    0,
+    1,
+    0,
+)
+PROCESS_LABEL_GOOD_DUPLICATES = (
+    """
+    label process_high
+    label process_high
+    cpus 12
+    """,
+    0,
+    2,
+    0,
+)
+PROCESS_LABEL_GOOD_AND_NONSTANDARD = (
+    """
+    label process_high
+    label process_extra_label
+    cpus 12
+    """,
+    1,
+    2,
+    0,
+)
+PROCESS_LABEL_NONSTANDARD = (
+    """
+    label process_extra_label
+    cpus 12
+    """,
+    0,
+    1,
+    0,
+)
+PROCESS_LABEL_NONSTANDARD_DUPLICATES = (
+    """
+    label process_extra_label
+    label process_extra_label
+    cpus 12
+    """,
+    0,
+    3,
+    0,
+)
+PROCESS_LABEL_NONE_FOUND = (
+    """
+    cpus 12
+    """,
+    0,
+    1,
+    0,
+)
+
+PROCESS_LABEL_TEST_CASES = [
+    PROCESS_LABEL_GOOD,
+    PROCESS_LABEL_NON_ALPHANUMERIC,
+    PROCESS_LABEL_GOOD_CONFLICTING,
+    PROCESS_LABEL_GOOD_DUPLICATES,
+    PROCESS_LABEL_GOOD_AND_NONSTANDARD,
+    PROCESS_LABEL_NONSTANDARD,
+    PROCESS_LABEL_NONSTANDARD_DUPLICATES,
+    PROCESS_LABEL_NONE_FOUND,
+]
+
+
+@pytest.mark.parametrize("lines,passed,warned,failed", PROCESS_LABEL_TEST_CASES)
+def test_modules_lint_check_process_labels(self, lines, passed, warned, failed):
```

Comment:
I think the test is not executed, should be added to `test_modules.py`, but I'm not sure how to do this when using the `@pytest.mark.parametrize` decorator.

Reply:
So it turns out you were right but also because we use `unittest.TestCase` I cannot parametrize that function. I've just added the test cases as a loop in the test for now. I think it's OK like this for now (to make the release) but we might want to refactor that in future.

Reply:
LGTM like this 🙂 
https://github.com/nf-core/tools/pull/2226 could be the way to refactor this in the future